File: CrmApp.Client/src/models/DTO/catalog-items/CatalogItemsListItemDTO.ts
import { AdditionalFieldValueDTO } from "../AdditionalFieldValueDTO";

export class CatalogItemsListItemDTO { 
    id: number = 0;
    name: string = "";
    code: string = "";
    description?: string;
    categoryId: number = 0;
    type: number = 0;
    unitName: string = "";
    billingUnitName: string = "";
    price: number = 0;
    isActive: boolean = true;
    vatRate: number = 0;
    currency: string = "";
    parentId?: number;
    technicalSupervisorId?: number;
    implementationManagerId?: number;
    additionalFieldValues: AdditionalFieldValueDTO[] = [];
}

File: CrmApp.Domain/DTO/CatalogItemsDTO/CatalogItemsListItemDTO.cs
using CrmApp.Domain.DTO.AdditionalFields;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CrmApp.Domain.DTO.CatalogItemsDTO
{
    public class CatalogItemsListItemDTO
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Code { get; set; } = string.Empty;
        public string? Description { get; set; }
        public int CategoryId { get; set; }
        public int Type { get; set; }
        public string UnitName { get; set; } = string.Empty;
        public string BillingUnitName { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public bool IsActive { get; set; }
        public decimal VatRate { get; set; }
        public string Currency { get; set; } = string.Empty;
        public int? ParentId { get; set; }
        public int? TechnicalSupervisorId { get; set; }
        public int? ImplementationManagerId { get; set; }
        public List<AdditionalFieldValueDTO> AdditionalFieldValues { get; set; } = [];
    }
}
